src/shared/utils/helpers/get-query-params.ts

```diff
@@ -1,17 +1,19 @@
 import { isBrowser } from "@utils/browser";
+import { GetSiteResponse } from "lemmy-js-client";
 
-export default function getQueryParams<
-  T extends Record<string, any>
->(processors: {
-  [K in keyof T]: (param: string) => T[K];
-}): T {
+export default function getQueryParams<T extends Record<string, any>>(
+  processors: {
+    [K in keyof T]: (param: string) => T[K];
+  },
+  site?: GetSiteResponse
+): T {
   if (isBrowser()) {
     const searchParams = new URLSearchParams(window.location.search);
 
     return Array.from(Object.entries(processors)).reduce(
       (acc, [key, process]) => ({
         ...acc,
-        [key]: process(searchParams.get(key)),
+        [key]: process(searchParams.get(key), site),
```

Comment:
Why would a GetSiteResponse necessarily have properties that should be query params?

Reply:
It's for implementing the fallback for listing type to the site-level setting set in the site admin. Would it make more sense to do something like return null in these callbacks and do the fallback on the home component?

Reply:
> Would it make more sense to do something like return null in these callbacks and do the fallback on the home component?

Yes. You don't want any of these props in the query params since no page uses them:

```typescript
export interface GetSiteResponse {
  site_view: SiteView;
  admins: Array<PersonView>;
  version: string;
  my_user?: MyUserInfo;
  all_languages: Array<Language>;
  discussion_languages: Array<LanguageId>;
  taglines: Array<Tagline>;
  custom_emojis: Array<CustomEmojiView>;
}
```

Reply:
Ok I've updated that.